core/libp2p/network/network.hpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_NETWORK_HPP
+#define KAGOME_NETWORK_HPP
+
+#include <outcome/outcome.hpp>
+
+#include "libp2p/connection/capable_connection.hpp"
+#include "libp2p/connection/stream.hpp"
+#include "libp2p/network/router.hpp"
+#include "libp2p/peer/peer_info.hpp"
+#include "libp2p/peer/protocol.hpp"
+
+namespace libp2p::network {
+
+  // Network is the interface used to connect to the outside world.
+  // It dials and listens for connections.
+  // it stores all connections
+  struct Network {
+    virtual ~Network() = default;
+
+    using connection_t = connection::CapableConnection;
+    using connptr_t = std::shared_ptr<connection_t>;
```

Comment:
Are we going to use connection_t type below? Do we really need it?
connptr_t is based on another naming convention.
maybe ConnectionPtr is better?

Reply:
It's better to decide such questions during implementation, I think. For now, it's hard to say, if this type is usable in this form or not, but the implementor of `Network` class will understand it for sure.